src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -961,13 +949,263 @@ def load_module(name, value):
                 f"Pipeline {pipeline_class} expected {expected_modules}, but only {passed_modules} were passed."
             )
 
-        # 5. Instantiate the pipeline
+        # 8. Instantiate the pipeline
         model = pipeline_class(**init_kwargs)
 
+        return_cached_folder = kwargs.pop("return_cached_folder", False)
         if return_cached_folder:
+            message = f"Passing `return_cached_folder=True` is deprecated and will be removed in `diffusers=0.17.0`. Please do the following instead: \n 1. Load the cached_folder via `cached_folder={cls}.download({pretrained_model_name_or_path})`. \n 2. Load the pipeline by loading from the cached folder: `pipeline={cls}.from_pretrained(cached_folder)`."
+            deprecate("return_cached_folder", "0.17.0", message, take_from=kwargs)
             return model, cached_folder
+
         return model
 
+    @classmethod
+    def download(cls, pretrained_model_name_or_path, **kwargs) -> Union[str, os.PathLike]:
```

Comment:
All this function does is to download the pipeline. Since this has become quite complex and also because it might be useful to have this as a standalone function it's factored out here (cc @keturn)